docs/develop/typescript/message-passing.mdx

```diff
@@ -196,107 +290,319 @@ await client.workflow.signalWithStart(yourWorkflow, {
+
+- **The workflow does not exist**:
+  You'll receive an [`common.WorkflowNotFoundError`](https://typescript.temporal.io/api/classes/common.WorkflowNotFoundError) error.
+
+
+### Problems when sending a Signal {#signal-problems}
+
+When using Signal, the two errors described above are the only errors that will result from your requests.
+
+For Queries and Updates, the client waits for a response from the Worker and therefore additional errors may occur during the handler Execution by the Worker.
+
+### Problems when sending an Update {#update-problems}
+
+When working with Updates, you may encounter these problems:
+
+- **No Workflow Workers are polling the Task Queue**:
+  Your request will be retried by the SDK Client indefinitely.
+
+- **Update failed**: You'll receive a [`client.WorkflowUpdateFailedError`](https://typescript.temporal.io/api/classes/client.WorkflowUpdateFailedError) exception.
+  There are two ways this can happen:
+
+    - The Update was rejected by an Update validator defined in the Workflow alongside the Update handler.
+
+    - The Update failed after having been accepted.
+
+    Update failures are like [Workflow failures](/references/failures#errors-in-workflows).
+    Issues that cause a Workflow failure in the main method also cause Update failures in the Update handler.
+    These might include:
+
+      - A failed Child Workflow
+      - A failed Activity (if the Activity retries have been set to a finite number)
+      - The Workflow author raising `ApplicationFailure`
+
+- **The handler caused the Workflow Task to fail**:
+  A [Workflow Task Failure](/references/failures#errors-in-workflows) causes the server to retry Workflow Tasks indefinitely. What happens to your Update request depends on its stage:
+    - If the request hasn't been accepted by the server, you receive a [`client.ServiceError`](https://typescript.temporal.io/api/classes/client.ServiceError) on which the `cause.code` attribute is [gRPC status code](https://grpc.io/docs/guides/status-codes/) 9 `FAILED_PRECONDITION` (after some retries).
+    - If the request has been accepted, it is durable.
+      Once the Workflow is healthy again after a code deploy, use an [`WorkflowUpdateHandle`](https://typescript.temporal.io/api/interfaces/client.WorkflowUpdateHandle) to fetch the Update result.
+
+- **The Workflow finished while the Update handler execution was in progress**:
+  You'll receive a [`client.ServiceError`](https://typescript.temporal.io/api/classes/client.ServiceError) on which the `cause.code` attribute is [gRPC status code](https://grpc.io/docs/guides/status-codes/) 5 `NOT_FOUND`.
+  This happens if the Workflow finished while the Update handler execution was in progress, for example because
+
+    - The Workflow was canceled or failed.
+
+    - The Workflow completed normally or continued-as-new and the Workflow author did not [wait for handlers to be finished](/encyclopedia/workflow-message-passing#finishing-message-handlers).
```

Comment:
Should we mention that is is ok for an update handler to complete immediately after wf function's completion, winthin the same WFT?

Reply:
I'd have to think more about that. One challenge is that our workflow programming abstraction hides the WFT concept from users, except for in the event history. Resolving as we've decided to address any remaining questions in follow-on work.